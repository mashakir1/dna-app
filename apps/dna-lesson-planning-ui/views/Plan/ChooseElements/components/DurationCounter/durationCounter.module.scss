.durationCounter {
  height: 110px;
  min-width: 110px;
  background-color: $colour_background_teal;
  border-radius: 1000px;
}
