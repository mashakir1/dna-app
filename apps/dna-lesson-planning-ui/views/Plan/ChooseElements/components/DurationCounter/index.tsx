// Imports
import { useMemo } from "react";
import useTranslation from "next-translate/useTranslation";

// Components
import { ClockIcon } from "components/Svg";

// Styles
import styles from "./durationCounter.module.scss";

export interface DurationCounterProps {
  minutes: number;
}

const DurationCounter: (props: DurationCounterProps) => JSX.Element = ({ minutes }) => {
  const { t } = useTranslation("lessonPlanning");
  const { hours, durationUnit, durationValue } = useMemo(() => {
    const hours = (minutes - (minutes % 60)) / 60;
    return {
      hours,
      durationUnit: hours > 0 ? t("hour-symbol") : t("minute-symbol"),
      durationValue: hours > 0 ? hours : minutes
    };
  }, [t, minutes]);

  return <div className={`${styles.durationCounter} flex column justify-center gap-xs items-center`}>
    <div className="flex gap-xs items-center">
      <p className="large"><b>{durationValue}</b></p>
      <p className="small">{durationUnit}</p>
      {hours > 0 && minutes % 60 > 0 && <p className="large"><b>{minutes % 60}</b></p>}
    </div>
    <ClockIcon />
  </div>;
};

export default DurationCounter;
