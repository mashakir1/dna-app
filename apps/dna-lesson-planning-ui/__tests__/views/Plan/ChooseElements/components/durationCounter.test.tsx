import { render, screen } from "@testing-library/react";
import DurationCounter from "views/Plan/ChooseElements/components/DurationCounter";

describe("DurationCounter", () => {
  it("renders minutes when no duration passed in", () => {
    const { container } = render(<DurationCounter minutes={0} />);

    expect(container).toMatchSnapshot();
    expect(screen.queryByText("0")).toBeTruthy();
    expect(screen.queryByText("minute-symbol")).toBeTruthy();
  });

  it("renders minutes when minutes are less than 60", () => {
    const { container } = render(<DurationCounter minutes={30} />);

    expect(container).toMatchSnapshot();
    expect(screen.queryByText("30")).toBeTruthy();
    expect(screen.queryByText("minute-symbol")).toBeTruthy();
  });

  it("renders hours when minutes are greater than 60", () => {
    const { container } = render(<DurationCounter minutes={90} />);

    expect(container).toMatchSnapshot();
    expect(screen.queryByText("1")).toBeTruthy();
    expect(screen.queryByText("hour-symbol")).toBeTruthy();
    expect(screen.queryByText("30")).toBeTruthy();
  });

  it("doesnt render minutes when exactly hours", () => {
    const { container } = render(<DurationCounter minutes={120} />);

    expect(container).toMatchSnapshot();
    expect(screen.queryByText("2")).toBeTruthy();
    expect(screen.queryByText("hour-symbol")).toBeTruthy();
    expect(screen.queryByText("0")).toBeFalsy();
  });
});
