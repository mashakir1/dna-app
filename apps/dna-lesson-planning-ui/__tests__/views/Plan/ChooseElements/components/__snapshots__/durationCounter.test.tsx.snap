// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DurationCounter doesnt render minutes when exactly hours 1`] = `
<div>
  <div
    class="durationCounter flex column justify-center gap-xs items-center"
  >
    <div
      class="flex gap-xs items-center"
    >
      <p
        class="large"
      >
        <b>
          2
        </b>
      </p>
      <p
        class="small"
      >
        hour-symbol
      </p>
    </div>
    <svg
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M12 4.63636C7.93318 4.63636 4.63636 7.93318 4.63636 12C4.63636 16.0668 7.93318 19.3636 12 19.3636C16.0668 19.3636 19.3636 16.0668 19.3636 12C19.3636 7.93318 16.0668 4.63636 12 4.63636ZM3 12C3 7.02944 7.02944 3 12 3C16.9706 3 21 7.02944 21 12C21 16.9706 16.9706 21 12 21C7.02944 21 3 16.9706 3 12ZM12 6.27273C12.4519 6.27273 12.8182 6.63904 12.8182 7.09091V11.4943L15.6386 12.9046C16.0428 13.1066 16.2066 13.5981 16.0045 14.0023C15.8024 14.4064 15.311 14.5702 14.9068 14.3682L11.6341 12.7318C11.3569 12.5932 11.1818 12.3099 11.1818 12V7.09091C11.1818 6.63904 11.5481 6.27273 12 6.27273Z"
        fill="#30CCD6"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</div>
`;

exports[`DurationCounter renders hours when minutes are greater than 60 1`] = `
<div>
  <div
    class="durationCounter flex column justify-center gap-xs items-center"
  >
    <div
      class="flex gap-xs items-center"
    >
      <p
        class="large"
      >
        <b>
          1
        </b>
      </p>
      <p
        class="small"
      >
        hour-symbol
      </p>
      <p
        class="large"
      >
        <b>
          30
        </b>
      </p>
    </div>
    <svg
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M12 4.63636C7.93318 4.63636 4.63636 7.93318 4.63636 12C4.63636 16.0668 7.93318 19.3636 12 19.3636C16.0668 19.3636 19.3636 16.0668 19.3636 12C19.3636 7.93318 16.0668 4.63636 12 4.63636ZM3 12C3 7.02944 7.02944 3 12 3C16.9706 3 21 7.02944 21 12C21 16.9706 16.9706 21 12 21C7.02944 21 3 16.9706 3 12ZM12 6.27273C12.4519 6.27273 12.8182 6.63904 12.8182 7.09091V11.4943L15.6386 12.9046C16.0428 13.1066 16.2066 13.5981 16.0045 14.0023C15.8024 14.4064 15.311 14.5702 14.9068 14.3682L11.6341 12.7318C11.3569 12.5932 11.1818 12.3099 11.1818 12V7.09091C11.1818 6.63904 11.5481 6.27273 12 6.27273Z"
        fill="#30CCD6"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</div>
`;

exports[`DurationCounter renders minutes when minutes are less than 60 1`] = `
<div>
  <div
    class="durationCounter flex column justify-center gap-xs items-center"
  >
    <div
      class="flex gap-xs items-center"
    >
      <p
        class="large"
      >
        <b>
          30
        </b>
      </p>
      <p
        class="small"
      >
        minute-symbol
      </p>
    </div>
    <svg
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M12 4.63636C7.93318 4.63636 4.63636 7.93318 4.63636 12C4.63636 16.0668 7.93318 19.3636 12 19.3636C16.0668 19.3636 19.3636 16.0668 19.3636 12C19.3636 7.93318 16.0668 4.63636 12 4.63636ZM3 12C3 7.02944 7.02944 3 12 3C16.9706 3 21 7.02944 21 12C21 16.9706 16.9706 21 12 21C7.02944 21 3 16.9706 3 12ZM12 6.27273C12.4519 6.27273 12.8182 6.63904 12.8182 7.09091V11.4943L15.6386 12.9046C16.0428 13.1066 16.2066 13.5981 16.0045 14.0023C15.8024 14.4064 15.311 14.5702 14.9068 14.3682L11.6341 12.7318C11.3569 12.5932 11.1818 12.3099 11.1818 12V7.09091C11.1818 6.63904 11.5481 6.27273 12 6.27273Z"
        fill="#30CCD6"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</div>
`;

exports[`DurationCounter renders minutes when no duration passed in 1`] = `
<div>
  <div
    class="durationCounter flex column justify-center gap-xs items-center"
  >
    <div
      class="flex gap-xs items-center"
    >
      <p
        class="large"
      >
        <b>
          0
        </b>
      </p>
      <p
        class="small"
      >
        minute-symbol
      </p>
    </div>
    <svg
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M12 4.63636C7.93318 4.63636 4.63636 7.93318 4.63636 12C4.63636 16.0668 7.93318 19.3636 12 19.3636C16.0668 19.3636 19.3636 16.0668 19.3636 12C19.3636 7.93318 16.0668 4.63636 12 4.63636ZM3 12C3 7.02944 7.02944 3 12 3C16.9706 3 21 7.02944 21 12C21 16.9706 16.9706 21 12 21C7.02944 21 3 16.9706 3 12ZM12 6.27273C12.4519 6.27273 12.8182 6.63904 12.8182 7.09091V11.4943L15.6386 12.9046C16.0428 13.1066 16.2066 13.5981 16.0045 14.0023C15.8024 14.4064 15.311 14.5702 14.9068 14.3682L11.6341 12.7318C11.3569 12.5932 11.1818 12.3099 11.1818 12V7.09091C11.1818 6.63904 11.5481 6.27273 12 6.27273Z"
        fill="#30CCD6"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</div>
`;
